client/src/main/java/io/pravega/client/connection/impl/FlowHandler.java

```diff
@@ -157,96 +160,18 @@ public AppendBatchSizeTracker getAppendBatchSizeTracker(final long requestID) {
-        Preconditions.checkNotNull(future, "future");
-        registeredFutureLatch.register(future);
-    }
-
     @Override
-    public void channelActive(ChannelHandlerContext ctx) throws Exception {
-        super.channelActive(ctx);
-        Channel ch = ctx.channel();
-        channel.set(ch);
-        log.info("Connection established with endpoint {} on channel {}", connectionName, ch);
-        ch.writeAndFlush(new WireCommands.Hello(WireCommands.WIRE_VERSION, WireCommands.OLDEST_COMPATIBLE_VERSION), ch.voidPromise());
-        registeredFutureLatch.release(null); //release all futures waiting for channel registration to complete.
-        // WireCommands.KeepAlive messages are sent for every network connection to a SegmentStore.
-        ScheduledFuture<?> old = keepAliveFuture.getAndSet(ch.eventLoop()
-                                                             .scheduleWithFixedDelay(keepAlive,
-                                                                                     KEEP_ALIVE_TIMEOUT_SECONDS,
-                                                                                     KEEP_ALIVE_TIMEOUT_SECONDS,
-                                                                                     TimeUnit.SECONDS));
-        if (old != null) {
-            old.cancel(false);
-        }
-    }
-
-    /**
-     * Invoke all the {@link ReplyProcessor#connectionDropped()} for all the registered flows once the
-     * connection is disconnected.
-     *
-     * @see io.netty.channel.ChannelInboundHandler#channelUnregistered(ChannelHandlerContext)
-     */
-    @Override
-    public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
-        ScheduledFuture<?> future = keepAliveFuture.get();
-        if (future != null) {
-            future.cancel(false);
-        }
-        channel.set(null);
-        log.info("Connection drop observed with endpoint {}", connectionName);
-        flowIdReplyProcessorMap.forEach((flowId, rp) -> {
-            try {
-                log.debug("Connection dropped for flow id {}", flowId);
-                rp.connectionDropped();
-            } catch (Exception e) {
-                // Suppressing exception which prevents all ReplyProcessor.connectionDropped from being invoked.
-                log.warn("Encountered exception invoking ReplyProcessor for flow id {}", flowId, e);
-            }
-        });
-        registeredFutureLatch.releaseExceptionally(new ConnectionClosedException());
-        super.channelUnregistered(ctx);
-    }
-
-    @Override
-    public void channelRead(ChannelHandlerContext ctx, Object msg) {
-        Reply cmd = (Reply) msg;
-        log.debug(connectionName + " processing reply {} with flow {}", cmd, Flow.from(cmd.getRequestId()));
-        recentMessage.set(true);
+    public void process(Reply cmd) {
+        log.debug("{} processing reply {} with flow {}", location, cmd, Flow.from(cmd.getRequestId()));
```

Comment:
What does `Flow.from` do? Most of the time we won't have `debug` logs enabled, but this will be evaluated each time. Is this something we could optimize by checking `log.isDebugEnabled()` beforehand?

Reply:
It is not expensive, but I added the check.